main.py

```diff
@@ -2,17 +2,55 @@
 import requests
 import pandas as pd
 from tqdm import tqdm
+import urllib
+
+
+def parse_book_url(url, title, author, format_='pdf'):
+    """
+    Given the original book URL, get the download URL and human-friendly
+    local filename for the given format ('pdf' or 'epub').
+
+    This function produces a URL but does not check that it is valid.
+    """
+
+    format_ = format_.lower()
+
+    # Replace percent-encoding with UTF-8 characters, replace "/book/..." URL
+    # path prefix with "/content/pdf/..." (or "/content/epub/...", etc. based
+    # on `format_`), and add format extension.
+    download_url = urllib.parse.unquote(url)
+    download_url = download_url.replace('book', 'content/' + format_)
```

Comment:
For the `epub` format, files are located at `download/epub`, not `content/epub`.
Correct handling:
```suggestion
    if format_ == "pdf":
        download_url = download_url.replace('book', 'content/' + format_)
    elif format_ == "epub":
        download_url = download_url.replace('book', 'download/' + format_)
    else:
        raise ValueError("Unsupported format: {}".format(format_))
```

Reply:
@niluxv Good catch, thanks!